File: position.cs
using System;

class Program
{
    // Метод для ввода числа и позиции
    static void Input(out string N, out int P)
    {
        Console.Write("Введите число N: ");
        N = Console.ReadLine();
        Console.Write("Введите позицию P: ");
        P = int.Parse(Console.ReadLine()); //преобразование строки в число 
    }

    // Метод для вывода символа на указанной позиции
    static void Output(string N, int P)
    {
        int length = N.Length; // длина строки
        if (P <= 0 || P > length)
        {
            // позиция должна быть больше нуля и меньше длины числа
            Console.WriteLine("Некорректная позиция P");
        }
        else
        {
            char result = N[length - P];
            Console.WriteLine(result);
        }
    }

    static void Main()
    {
        string N;
        int P;

        Input(out N, out P);
        Output(N, P);
    }
}

File: position.rs
use std::io; // Модуль ввода-вывода, который позволяет читать данные с клавиатуры

// Функция для ввода числа и позиции
fn input() -> (String, usize) { // Возвращает кортеж (String, usize)
    println!("Введите число N: ");
    let mut n = String::new(); // Пустая изменяемая (mut) строка n
    io::stdin()
        .read_line(&mut n)
        .expect("Ошибка чтения строки"); // Читает введённую строку и сохраняет её в n
    let n = n.trim().to_string(); // Удаляет пробелы и символы перевода строки (\n)

    println!("Введите позицию P: ");
    let mut p = String::new();
    io::stdin()
        .read_line(&mut p)
        .expect("Ошибка чтения строки");
    let p: usize = p.trim()
        .parse()
        .expect("Ожидалось целое число"); // Преобразует строку в число (usize)

    (n, p)
}

// Функция для вывода символа на указанной позиции
fn output(n: &str, p: usize) {
    if p == 0 || p > n.len() {
        println!("Некорректная позиция P");
    } else {
        let result = n.chars()
            .nth(n.len() - p)
            .unwrap(); // Получаем символ на нужной позиции
        println!("{}", result);
    }
}

fn main() {
    let (n, p) = input();
    output(&n, p);
}